### PR TITLE
replicated functionality of Makefile in Taskfile

### PR DESCRIPTION
**Pull request by saadatq**
Comparing to Makefile, taskfile felt a bit easier to learn. But since most of the information needed for taskfile is at https://taskfile.dev/usage/, it simplifies the process of learning taskfile. The advantage of taskfile is its readability and simple syntaxes, making it easier for newer members to edit and change the file. I found that taskfile is able to run multiple tasks in parallel but I have not been able to try it out.

Installing taskfile was a simple process by running: choco install go-task
- https://taskfile.dev/installation

### CODE REVIEW

File: firmware/Taskfile.dist.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  GENERATOR: -G 'Unix Makefiles'
+
+tasks:
+  project_name_fix:
+    cmds:
+      - echo 
+  build:
+    cmds:
+      - |
+        read -p "Enter the build: " BUILD
```
Comment (BlakeFreer) [inline firmware/Taskfile.dist.yml:13]: The "build" directory is fixed, it should not be an input. Move it to an env variable

**Comment by BlakeFreer** [comment]
I cannot build a project. For example:

```bash
$ task build
Enter the build: build
Enter the project: EV5/LVController
Enter the platform: cli
 is not a directoryepos/racecar/firmware/build

task: Failed to run task "build": exit status 1
```

How should I use it? The equivalent makefile command is `make build PROJECT=EV5/LVController PLATFORM=cli`

**Comment by BlakeFreer** [comment]
I'm concerned about the repetition of code in this file. The following lines are repeated (at least in-part) in every sub task. This will make it hard to maintain.

If you take the arguments from the command line (rather than prompting, see earlier comment) then you may be able to set these variables globally and use them in any subtask.

```yaml
      - |
        read -p "Enter the build: " BUILD 
        read -p "Enter the project: " PROJECT 
        read -p "Enter the platform: " PLATFORM

        BUILD_DIR="$BUILD/$PROJECT/$PLATFORM"
        PROJECT_NAME=$(echo "$PROJECT" | tr '/' '-')
        COMPILE_COMMANDS_DEST: '$BUILD/compile_commands.json'
        PLATFORM_DIR: "projects/$PROJECT/platforms/$PLATFORM"
```

**Review by BlakeFreer — CHANGES_REQUESTED** [review]

File: firmware/Taskfile.dist.yml (hunk 1 above)
Comment (saadatq) [reply firmware/Taskfile.dist.yml]: I made the changes in the latest push in my branch

**Comment by BlakeFreer** [comment]
Qasim and I decided that `task` is not a cleaner solution than `make`. We will close this PR without merging and will stay with `make` for now.

**Pull request closed without merging**